### PR TITLE
[op normalization] add NormalizedFragmentSpread and NormalizedFragment

### PR DESCRIPTION
**Pull request by dariuszkuc**
related: https://github.com/apollographql/federation-next/issues/104

### CODE REVIEW

File: src/query_plan/operation.rs [hunk 1]
```diff
@@ -31,7 +31,15 @@ pub enum NormalizedSelectionKey {
     Field {
         // field alias (if specified) or field name in the resulting selection set
         response_name: Name,
-        // directive applied on the field
+        // directives applied on the field
+        directives: DirectiveList,
+        // unique label/counter used to distinguish fields that cannot be merged
+        label: i32,
```
Comment (lrlna) [inline src/query_plan/operation.rs:37]: is an i32 enough? not exactly sure how we use this, but do we think we need an atomic counter here? or will that be an overkill for its usecase?

Comment (SimonSapin) [reply src/query_plan/operation.rs:39]: This is only unique within one document, right? I’d expect any document to reach the parser’s token limit well before it contains two billion selections.

File: src/query_plan/operation.rs [hunk 2]
```diff
@@ -406,6 +437,9 @@ fn merge_selections(
                             }
                         }
                     }
+                    NormalizedSelection::NormalizedFragmentSpread(_) => {
+                        // nothing to do fragment spread is already part of the selection set
```
Comment (sachindshinde) [inline src/query_plan/operation.rs:455]: nit: Might be nice to copy over [the relevant comment from Sylvain here](https://github.com/apollographql/federation/blob/f69a0694b95e99c22f0d5f7bc6df1b46568b2a8d/internals-js/src/operations.ts#L2182-L2185).

File: src/query_plan/operation.rs (hunk 1 above)
Comment (sachindshinde) [reply src/query_plan/operation.rs:39]: For now it's unique within the document. That will change once https://github.com/apollographql/federation-next/issues/103 is implemented (but I think the switch to `i64` can be put off till then).

**Review by sachindshinde — APPROVED** [review]

File: src/query_plan/operation.rs (hunk 1 above)
Comment (lrlna) [reply src/query_plan/operation.rs:39]: ^ sgtm

**Pull request merged**